src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -484,10 +518,21 @@ public void TestEnvironmentVariables_Environment_DataRoundTrips()
 
             string envVarResultKey = "";
             string envVarResultValue = "";
-            foreach(KeyValuePair<string, string> entry in psi.EnvironmentVariables)
+
+            // .NET Core uses a Dictionary<string, string>, but the full .NET Framework uses
+            // StringDictionary. These collections return different types from GetEnumerator.
```

Comment:
Unfortunately not, I think:
```cs
Assert.Equal(psi.Environment, psi.EnvironmentVariables);
```

fails to compile:
```cs
Assert.Equal(psi.Environment, (IEnumerable)psi.EnvironmentVariables);
```
fails with NotEqual.

```cs
Assert.Equal(psi.Environment, psi.EnvironmentVariables.Cast<KeyValuePair<string, string>>());
```

works with netcoreapp but fails with an InvalidOperationException with netfx